### PR TITLE
Fixes #31158 - fix auth for CVVs controller

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/models/katello/authorization/content_view_version.rb [hunk 1]
```diff
@@ -10,9 +10,32 @@ def all_hosts_editable?(lifecycle_environment)
     end
 
     module ClassMethods
+      def with_content_view_scope(scope)
+        joins(:content_view).where("#{Katello::ContentView.table_name}.id" => ::Katello::ContentView.send(scope))
```
Comment (jturel) [inline app/models/katello/authorization/content_view_version.rb:14]: If you include the Authorizable module in here you can do this: `joins_authorized(::Katello::ContentView, :view_content_views)` for example

**Comment by jturel** [comment]
left one note on authorization, but all else looks good. will test if you end up making the suggested change

File: app/models/katello/authorization/content_view_version.rb (hunk 1 above)
Comment (jturel) [reply app/models/katello/authorization/content_view_version.rb:14]: @jlsherrill maybe you'd prefer the merge approach? `joins(:content_view).merge(Katello::ContentView.readable)` for example

**Comment by jturel** [comment]
Needs a rebase now :-/

File: app/models/katello/authorization/content_view_version.rb (hunk 1 above)
Comment (jlsherrill) [reply app/models/katello/authorization/content_view_version.rb:14]: i'm hesitant to add Authorizable due to the check_permissions_after_save hook it adds, but i like .merge()

**Comment by jlsherrill** [comment]
updated and rebased!

**Comment by jturel** [comment]
[test katello]

**Review by jturel — APPROVED** [review]
I tried to break it but I couldn't! APJ

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**